apps/passport-server/src/services/rollbarService.ts

```diff
@@ -24,7 +24,14 @@ export class RollbarService {
 
   public reportError(e: any): void {
     if (e instanceof Error) {
-      this.rollbar.error(e, e.cause as object | undefined);
+      const errorChain: (Error | { rootCause: string })[] = [e];
+      let currentError = e;
+      while (currentError.cause && currentError.cause instanceof Error) {
+        errorChain.push(currentError.cause as Error);
+        currentError = currentError.cause;
+      }
+      errorChain.push({ rootCause: (errorChain.at(-1) as Error).message });
+      this.rollbar.error(...errorChain);
```

Comment:
I want to be sure this code never throws. Briefly glancing at it, the various `as` are kind of scary, and the `errorChain.at(-1)` is a bit scary.

Reply:
The `as` comes after a check for the existence of a `.cause` field, but yes, there could be non-`Error` things which have `.cause` fields, so I will make the check more rigorous.